data/src/main/kotlin/com/bitwarden/data/repository/util/EnvironmentUrlDataJsonExtensions.kt

```diff
@@ -2,73 +2,93 @@ package com.bitwarden.data.repository.util
-        ?: DEFAULT_EVENTS_URL
+    get() = when (this.environmentRegion) {
+        EnvironmentRegion.UNITED_STATES -> DEFAULT_US_EVENTS_URL
+        EnvironmentRegion.EUROPEAN_UNION -> DEFAULT_EU_EVENTS_URL
+        EnvironmentRegion.SELF_HOSTED -> {
+            this.events.sanitizeUrl
+                ?: this.base.sanitizeUrl?.let { "$it/events" }
+                ?: DEFAULT_US_EVENTS_URL
+        }
+    }
 
 /**
  * Returns the base identity URL or the default value if one is not present.
  */
 val EnvironmentUrlDataJson.baseIdentityUrl: String
-    get() = this.identity.sanitizeUrl
-        ?: this.base.sanitizeUrl?.let { "$it/identity" }
-        ?: DEFAULT_IDENTITY_URL
+    get() = when (this.environmentRegion) {
+        EnvironmentRegion.UNITED_STATES -> DEFAULT_US_IDENTITY_URL
+        EnvironmentRegion.EUROPEAN_UNION -> DEFAULT_EU_IDENTITY_URL
+        EnvironmentRegion.SELF_HOSTED -> {
+            this.identity.sanitizeUrl
+                ?: this.base.sanitizeUrl?.let { "$it/identity" }
+                ?: DEFAULT_US_IDENTITY_URL
+        }
+    }
 
 /**
  * Returns the base web vault URL. This will check for a custom [EnvironmentUrlDataJson.webVault]
  * before falling back to the [EnvironmentUrlDataJson.base]. This can still return null if both are
  * null or blank.
  */
 val EnvironmentUrlDataJson.baseWebVaultUrlOrNull: String?
-    get() = this.webVault.sanitizeUrl
-        ?: this.base.sanitizeUrl
+    get() = when (this.environmentRegion) {
+        EnvironmentRegion.UNITED_STATES -> DEFAULT_US_WEB_VAULT_URL
+        EnvironmentRegion.EUROPEAN_UNION -> DEFAULT_EU_WEB_VAULT_URL
+        EnvironmentRegion.SELF_HOSTED -> this.webVault.sanitizeUrl ?: this.base.sanitizeUrl
+    }
 
 /**
  * Returns the base web vault URL or the default value if one is not present.
  *
  * See [baseWebVaultUrlOrNull] for more details.
  */
 val EnvironmentUrlDataJson.baseWebVaultUrlOrDefault: String
-    get() = this.baseWebVaultUrlOrNull ?: DEFAULT_WEB_VAULT_URL
+    get() = this.baseWebVaultUrlOrNull ?: DEFAULT_US_WEB_VAULT_URL
 
 /**
  * Returns the base web send URL or the default value if one is not present.
  */
 val EnvironmentUrlDataJson.baseWebSendUrl: String
-    get() =
-        this
-            .baseWebVaultUrlOrNull
-            ?.let {
-                // Only on US Cloud we should use the default web send URL
-                // On all other server instances we should use the base web send URL
-                if (it == DEFAULT_WEB_VAULT_URL) {
-                    DEFAULT_WEB_SEND_URL
-                } else {
-                    "$it/#/send/"
-                }
-            }
-            ?: DEFAULT_WEB_SEND_URL
+    get() = when (this.environmentRegion) {
+        EnvironmentRegion.UNITED_STATES -> DEFAULT_US_WEB_SEND_URL
+        EnvironmentRegion.EUROPEAN_UNION,
+        EnvironmentRegion.SELF_HOSTED,
+            -> this.baseWebVaultUrlOrNull?.let { "$it/#/send/" } ?: DEFAULT_US_WEB_SEND_URL
```

Comment:
⛏️ Not sure if this would affect anything but wanted to bring this up: In the default send URL it's not adding a `/` at the end and we're not doing that either in iOS. Perhaps that needs to be removed?
```suggestion
            -> this.baseWebVaultUrlOrNull?.let { "$it/#/send" } ?: DEFAULT_US_WEB_SEND_URL
```

Reply:
The way the app currently works requires this trailing forward slash. I would rather not change that in this PR.